### ISSUE TITLE
Random crashes on Linux

### ISSUE THREAD

**Issue by BartT78** [gridcoin-community/Gridcoin-Research#907]
OS: Kubuntu 16.04
Architecture: amd64
Kernel release: 4.13.0-32-generic

Client version: v3.7.5.0-unk-317
OpenSSL version: OpenSSL 1.0.2g  1 Mar 2016
Boost version: 1.58.0

Installed from repository: ppa:gridcoin/gridcoin-stable
Package version: 3.7.5.0-13~ubuntu16.04.1

Application crashes randomly with various error messages.

Sample error messages:

```
Error in `/usr/bin/gridcoinresearch': free(): invalid size: 0x00007f4c9cb2b8b0
```
```
Error in `/usr/bin/gridcoinresearch': free(): invalid next size (normal): 0x00007f7c31503f00
```

**Comment by iFoggz** [comment]
neither at the same time either eh? never a good sign as should never be invalid

**Comment by BartT78** [comment]
I had one application crash yesterday and at least three today. Messages come from the last two  crashes today.

**Comment by iFoggz** [comment]
can u tell me more about ur setup? so  can try to duplicate this in gdb with debug symbols in the application?

staking?
full node?

**Comment by BartT78** [comment]
1. BOINC client connected to grcpool.com

boinc-client (version: 7.6.31+dfsg-6ubuntu1)
boinc-manager (version: 7.6.31+dfsg-6ubuntu1)
both installed from Ubuntu repositories

2.  Gridcoin client configured in investor mode.

gridcoinresearch.conf

```
email=INVESTOR
addnode=grcnode.tahvok.com
addnode=grcexplorer.neuralminer.io
addnode=grcnode01.neuralminer.io
addnode=grcnode02.neuralminer.io
addnode=grcnode03.neuralminer.io
addnode=grcnode04.neuralminer.io
addnode=grcnode05.neuralminer.io
addnode=gridcoin.bunnyfeet.fi
addnode=gridcoin.certic.info
addnode=gridcoin.crypto.fans
addnode=gridcoin.hopto.org
addnode=ils.gridcoin.co.il
addnode=la.grcnode.co.uk
addnode=london.grcnode.co.uk
addnode=miami.grcnode.co.uk
addnode=node.gridcoin.network
addnode=node.gridcoin.us
addnode=node.gridcoinapp.xyz
addnode=node1.chick3nman.com
addnode=nuad.de
addnode=quebec.gridcoin.co.il
addnode=seeds.gridcoin.ifoggz-network.xyz
addnode=singapore.grcnode.co.uk
addnode=toronto01.gridcoin.ifoggz-network.xyz
addnode=vancouver01.gridcoin.ifoggz-network.xyz
addnode=www.grcpool.com
maxoutboundconnections=24
server=1
daemon=1
rpcallowip=127.0.0.1
rpcuser=gridcoinrpc
rpcpassword=[REMOVED BEFORE SENDING TO GITHUB]
publickey[REMOVED BEFORE SENDING TO GITHUB]
privatekey[REMOVED BEFORE SENDING TO GITHUB]
```

3. Network configuration

Client installed on the computer with private static IP behind the router with public static IP and with UPnP enabled.

**Comment by iFoggz** [comment]
that's a tuff one. would of been more helpful with a debug version and running in gdb. for now upgrade to 3.7.7.0

**Comment by denravonska** [comment]
That crash is most likely the encrypt/decrypt crash fixed in 3.7.7.0.

**Comment by denravonska** [comment]
Closing as fixed for now. Please reopen if you are still experiencing it.
